### PR TITLE
deployerPod: Clean-up desired replicas annotation for a complete deployment

### PR DESCRIPTION
**Pull request by kargakis**
@ironcladlou PTAL

cc: @smarterclayton

### CODE REVIEW

File: pkg/deploy/controller/deployerpod/controller.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ func (c *DeployerPodController) Handle(pod *kapi.Pod) error {
 	case kapi.PodSucceeded:
 		// Detect failure based on the container state
 		nextStatus = deployapi.DeploymentStatusComplete
+		delete(deployment.Annotations, deployapi.DesiredReplicasAnnotation)
```
Comment (ironcladlou) [inline pkg/deploy/controller/deployerpod/controller.go]: `nextStatus` is assigned to complete optimistically; the `for` loop immediately following might set it to failed, in which case the annotation should be preserved.

Let's add test coverage for this state.

Comment (kargakis) [inline pkg/deploy/controller/deployerpod/controller.go]: all done

**Comment by ironcladlou** [comment]
LGTM!

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**